### PR TITLE
add typemeta to kustomization

### LINKED ISSUE #144: Version kustomization.yaml files
I'm hoping to do some parsing of the kustomization.yaml files and it would be helpful if the config was versioned. Maybe kubernetes style?

### PR DESCRIPTION
**Pull request by Liujingfang1**
Fix #144

### CODE REVIEW

File: pkg/commands/kustomizationfile.go [hunk 1]
```diff
@@ -203,6 +203,9 @@ func isCommentOrBlankLine(line []byte) bool {
 // If the field wasn't in the original kustomization.yaml file or wasn't added,
 // an empty []byte is returned.
 func marshalField(field string, kustomization *types.Kustomization) ([]byte, error) {
+	if field == "apiVersion" {
```
Comment (mengqiy) [inline pkg/commands/kustomizationfile.go]: Can you add a comment explaining why you need to do this? I don't quite understand.

File: pkg/commands/kustomizationfile.go [hunk 2]
```diff
@@ -35,8 +35,8 @@ import (
 )
 
 var (
-	kustomizationFields = []string{"resources", "bases", "namePrefix", "namespace", "crds", "commonLabels", "commonAnnotations", "patches", "configMapGenerator", "secretGenerator", "vars", "imageTags"}
-	recognizedFields    = regexp.MustCompile("^(" + strings.Join(kustomizationFields, "|") + "):")
+	kustomizationFields = []string{"apiVersion", "kind", "resources", "bases", "namePrefix", "namespace", "crds", "commonLabels", "commonAnnotations", "patches", "configMapGenerator", "secretGenerator", "vars", "imageTags"}
```
Comment (monopole) [inline pkg/commands/kustomizationfile.go]: please add linebreaks for readability, e.g.
```
kustomizationFields = []string{
    "resources",
    "bases",
```

File: pkg/commands/kustomizationfile.go [hunk 3]
```diff
@@ -35,8 +35,8 @@ import (
...
+	recognizedFields    = regexp.MustCompile("^(" + "(?i)" + strings.Join(kustomizationFields, "|") + "):")
```
Comment (monopole) [inline pkg/commands/kustomizationfile.go]: Please add a comment explaining (?i)

File: pkg/commands/kustomizationfile.go (hunk 1 above)
Comment (monopole) [reply pkg/commands/kustomizationfile.go]: ditto

Comment (Liujingfang1) [reply pkg/commands/kustomizationfile.go]: In current `marshalField` function, the filed is assumed to be in a pattern that the correct field name can be get by `strings.Title`, such as `kind` --> `Kind`.  However, `apiVersion` doesn't follow this pattern. Let me know if the assumption is to strong. 

In addition, I found that `KIND` or `Kind` are supported by yaml. So I assume we also need to support them when we parse the kustomization.yaml file. That's why I added a `(?i)` in the regexp.

Comment (Liujingfang1) [reply pkg/commands/kustomizationfile.go]: I refactored the code for finding the matching field. See if it looks better.

File: pkg/commands/kustomizationfile.go [hunk 4]
```diff
@@ -129,20 +144,20 @@ func stringInSlice(str string, list []string) bool {
...
-			currentfield = string(fields[1])
-			mf.originalFields = append(mf.originalFields, &commentedField{field: currentfield, comment: bytes.Join(comments, []byte(``))})
-			comments = [][]byte{}
-		} else if len(comments) > 0 {
-			mf.originalFields[len(mf.originalFields)-1].appendComment(bytes.Join(comments, []byte(``)))
-			comments = [][]byte{}
+		} else {
+			matched, field := findMatchedField(line)
+			if matched {
+				mf.originalFields = append(mf.originalFields, &commentedField{field: field, comment: bytes.Join(comments, []byte(``))})
```
Comment (monopole) [inline pkg/commands/kustomizationfile.go]: Please add some line feeds and use a helper to increase readbility.

eg 
```
func squash(x [][]byte) []byte {
  return bytes.Join(x, []byte(``))
}
```

File: pkg/commands/kustomizationfile_test.go [hunk 5]
```diff
@@ -158,7 +160,7 @@ vars:
     kind: Service
     name: my-service
 
-bases:
+BASES:
```
Comment (monopole) [inline pkg/commands/kustomizationfile_test.go]: so this just tests case insensitivity?  nice.

Comment (Liujingfang1) [reply pkg/commands/kustomizationfile_test.go]: yes

File: pkg/commands/kustomizationfile.go (hunk 4 above)
Comment (Liujingfang1) [reply pkg/commands/kustomizationfile.go]: sounds good. Will add it.

Comment (Liujingfang1) [reply pkg/commands/kustomizationfile.go]: done

**Pull request merged**